README.md

```diff
@@ -1 +1,88 @@
-# visenze-tracking-android
\ No newline at end of file
+# visenze-tracking-android
+
+Android SDK for Visenze data tracking
+
+
+## 1. Overview
+
+Visenze data tracking sdk is an open source software to send user defined data tracking event to visenze for performance monitoring and analytic purpose.
+
+## 2. Setup and nitialization
+
+To initialize sdk, extend Application and initialize the sdk with application context and request code.
+
+In manifest set the MyApplication as application name
+
+```
+    <application
+        android:name=".MyApplication"
+        ...
+    </application>
+```
+
+
+```java
+
+public class MyApplication extends Application {
+
+    private static VisenzeAnalytics dataTracking;
+    private static Tracker tracker;
+    @Override
+    public void onCreate() {
+        super.onCreate();
+
+        dataTracking = VisenzeAnalytics.getInstance(this);
+    }
+
+    synchronized public Tracker getDefaultTracker() {
+        if(tracker == null) {
+            String code = "YOUR CODE";
+            tracker = dataTracking.newTracker(code, false);
+        }
+        return tracker;
+    }
+}
+
+```
+
+please contact visenze to get a code to uniquely identify your application.
+
+You can also set a uid to data tracking sdk. If the uid is not set. By default, the sdk will generate unique id to identify the customer.
+in addition, you can also pass device information to sdk to help visenze better understand the customer and improve analytics.
+
+```java
+
+    Tracker tracker = ((MyApplication) getApplication()).getDefaultTracker();
+    DeviceData deviceData = new DeviceData();
+    deviceData.setDidmd5("HASHED_DEVICE_ID");
+    tracker.setDeviceData(deviceData);
+
+```
+
+## 3. Send events
+
+To create event,  use Event.createXXXEvent() with input parameters to create event.
+
+```
+    Event e1 = Event.createViewEvent();
```

Comment:
we need to explain the 5 default events and examples in more details. But this can be done later when we kick off the Release process with testing.